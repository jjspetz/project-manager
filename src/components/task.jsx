import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ItemTypes } from '../constants/constants.js';
import { DragSource } from 'react-dnd';

// class Task extends Component {
//
//   render() {
//     return (
//       <li key={this.props.task} draggable='true' >{this.props.task}</li>
//     );
//   }
// }
//
// export default Task


const taskSource = {
  beginDrag(props) {
    return {
      id: props.task.task,
      column: props.task.column
    };
  }
};

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  }
}

class Task extends Component {
  render() {
    const { connectDragSource, isDragging } = this.props;
    return connectDragSource(
      <li key={this.props.task.task} style={{
        opacity: isDragging ? 0.5 : 1,
        cursor: 'move'
      }}>
        {this.props.task.task}
      </li>
    );
  }
}

Task.propTypes = {
  connectDragSource: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired
};

export default DragSource(ItemTypes.TASK, taskSource, collect)(Task);
