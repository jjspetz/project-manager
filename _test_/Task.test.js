import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import { Task } from '../src/components/task.jsx';

const faux = el => el;
const taskName = 'testing'
const wrapper = shallow(<Task task={taskName}
                              connectDragSource={faux}
                              isDragging />);

describe('Task', () => {
  it('should render', () => {
    expect(wrapper).toBeTruthy();
  });
  it('should contain a div', () => {
    expect(wrapper.find('div')).toBeTruthy();
  });
  it('should contain a li with the task', () => {
    expect(wrapper.find('div')).toBeTruthy();
    expect(wrapper.instance().props.task).toContain(taskName);
  });
  it('should match snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
