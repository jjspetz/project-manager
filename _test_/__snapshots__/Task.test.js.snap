// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should match snapshot 1`] = `
<div>
  <Paper
    circle={false}
    rounded={true}
    style={
      Object {
        "padding": 10,
        "width": "92%",
      }
    }
    transitionEnabled={true}
    zDepth={1}
  >
    <li
      style={
        Object {
          "cursor": "move",
          "opacity": 0.5,
        }
      }
    />
  </Paper>
</div>
`;
